src/gc/gc.cpp

```diff
@@ -5715,18 +5711,29 @@ void gc_mechanisms::record (gc_history_global* history)
 //as opposed to concurrent heap verification
 void gc_heap::fix_youngest_allocation_area (BOOL for_gc_p)
 {
-    assert (alloc_allocated);
-    alloc_context* acontext = generation_alloc_context (youngest_generation);
-    dprintf (3, ("generation 0 alloc context: ptr: %Ix, limit %Ix",
-                 (size_t)acontext->alloc_ptr, (size_t)acontext->alloc_limit));
-    fix_allocation_context (acontext, for_gc_p, get_alignment_constant (TRUE));
-    if (for_gc_p)
+    alloc_context *acontext;
+
+    if (!using_thread_alloc_contexts())
     {
-        acontext->alloc_ptr = alloc_allocated;
-        acontext->alloc_limit = acontext->alloc_ptr;
+        acontext = global_allocation_context;
+        assert(acontext != nullptr);
+        dprintf (3, ("EE-owned gen 0 alloc context %Ix: ptr %Ix, limit: %Ix",
+              (size_t)acontext, (size_t)acontext->alloc_ptr, (size_t)acontext->alloc_limit));
     }
-    heap_segment_allocated (ephemeral_heap_segment) =
-        alloc_allocated;
+    else
+    {
+        acontext = generation_alloc_context (youngest_generation);
```

Comment:
>acontext = generation_alloc_context (youngest_generation); [](start = 8, length = 58)

we no longer use the gen0 alloc context in the allocation path so you don't need to fix it. you should assert that gen0's alloc context's limit and ptr are both 0 here. we only need to call GcEnumAllocContexts.